src/Essentials/src/Geolocation/Geolocation.android.cs

```diff
@@ -135,13 +145,17 @@ public async Task<bool> StartListeningForegroundAsync(GeolocationListeningReques
 		{
 			ArgumentNullException.ThrowIfNull(request);
 
+			if (LocationManager == null)
+				throw new FeatureNotSupportedException("Android LocationManager is not available");
+
 			if (IsListeningForeground)
 				throw new InvalidOperationException("Already listening to location changes.");
 
 			await Permissions.EnsureGrantedOrRestrictedAsync<Permissions.LocationWhenInUse>();
 
 			var enabledProviders = LocationManager.GetProviders(true);
-			var hasProviders = enabledProviders.Any(p => !ignoredProviders.Contains(p));
+			var hasProviders = enabledProviders != null &&
+				enabledProviders.Any(p => !ignoredProviders.Contains(p));
```

Comment:
```suggestion
			var hasProviders = enabledProviders?.Any(p => !ignoredProviders.Contains(p)) == true;
```

Reply:
it seems `hasProviders` is changed to type `bool?`, so instead I suggest:

    bool hasProviders = enabledProviders?.Any(p => !ignoredProviders.Contains(p)) ?? false;

Reply:
As commented below, maybe it's best to keep the more readable null check then?

Reply:
agreed :)